### PR TITLE
feat: auto-fix pack 2 — 8 additional unsafe auto-fix rules (#317)

### LINKED ISSUE #317: Auto-fix pack 2: additional auto-fix opportunities (8 rules)
## Summary

Additional auto-fix opportunities discovered during analysis of all 113 non-autofix rules. These 8 rules have validators already implemented and follow existing fix patterns, making them straightforward to add.

### Tier 1 — Trivial (identical patterns already exist)

| Rule | Fix | Safety | Pattern | Complexity |
|------|-----|--------|---------|------------|
| CC-SK-005 | Replace invalid `agent` type with `general-purpose` default | UNSAFE | Same as CC-SK-001/CC-SK-002 (`frontmatter_value_byte_range`) | Trivial (4 lines) |
| CC-AG-012 | Replace `bypassPermissions` with `default` | UNSAFE | Same as CC-AG-004 (`frontmatter_value_byte_range`) | Trivial (3 lines) |
| CUR-002 | Insert missing frontmatter template `---\ndescription:\nglobs:\n---\n` | UNSAFE | `Fix::insert(0, ...)` | Trivial (3 lines) |

### Tier 2 — Easy (small helper needed or pattern adaptation)

| Rule | Fix | Safety | Pattern | Complexity |
|------|-----|--------|---------|------------|
| CDX-001 | Replace invalid `approvalMode` with closest valid | UNSAFE | New `find_toml_string_value_span` helper | Easy (10 lines) |
| CDX-002 | Replace invalid `fullAutoErrorMode` with closest valid | UNSAFE | Reuses CDX-001 helper | Easy (4 lines after CDX-001) |
| OC-001 | Replace invalid `share` mode with closest valid | UNSAFE | Copy `find_unique_json_string_value_span` from mcp.rs | Easy (10 lines) |
| CC-HK-016 | Replace unknown hook `type` with closest valid (`command`/`prompt`/`agent`) | UNSAFE | Fuzzy match on JSON value, `find_unique_json_key_value_span` exists | Easy (12 lines) |
| COP-002 | Insert missing frontmatter for scoped instruction files | UNSAFE | `Fix::insert(0, ...)` with `applyTo` template | Easy (8 lines) |

## Implementation Notes

- All fixes in this pack are UNSAFE (require user verification) since they involve defaults or fuzzy matching
- CDX-001/CDX-002 need a new `find_toml_string_value_span(content, key)` helper for TOML files
- OC-001 can reuse/copy the `find_unique_json_string_value_span` helper from `mcp.rs`
- CC-AG-012 is debatable — removing `bypassPermissions` significantly changes agent behavior. Mark as unsafe with clear fix description.
- CUR-002 and COP-002 insert template frontmatter that the user still needs to fill in
- Update `rules.json` (`"autofix": true`, `"fix_safety": "unsafe"`) and `VALIDATION-RULES.md` for each rule
- Add i18n `fix:` strings to `locales/en.yml`

## Key Files

| File | Rules |
|------|-------|
| `crates/agnix-core/src/rules/skill/mod.rs` | CC-SK-005 (line ~643) |
| `crates/agnix-core/src/rules/agent.rs` | CC-AG-012 (line ~707) |
| `crates/agnix-core/src/rules/cursor.rs` | CUR-002 (line ~153) |
| `crates/agnix-core/src/rules/codex.rs` | CDX-001 (line ~86), CDX-002 (line ~117) |
| `crates/agnix-core/src/rules/opencode.rs` | OC-001 (line ~49) |
| `crates/agnix-core/src/rules/hooks/helpers.rs` | CC-HK-016 (line ~546) |
| `crates/agnix-core/src/rules/copilot.rs` | COP-002 (line ~128) |

## Dependencies

- #301 (Auto-fix pack 1) — not a hard dependency but should be done first to establish patterns for CDX/OC rules

## References

- Existing auto-fix patterns in `crates/agnix-core/src/rules/skill/mod.rs`, `agent.rs`, `mcp.rs`
- `crates/agnix-core/src/fixes.rs` (fix application engine)

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Adds UNSAFE auto-fix capabilities to 8 existing validation rules, bringing the total auto-fixable rules from 40 to 48 (33%).

### Rules with new auto-fixes

| Rule | Fix | Safety |
|------|-----|--------|
| CC-SK-005 | Replace invalid agent type with `general-purpose` | unsafe |
| CC-AG-012 | Replace `bypassPermissions` with `default` | unsafe |
| CUR-002 | Insert template frontmatter (description + globs) | unsafe |
| COP-002 | Insert template frontmatter (applyTo) | unsafe |
| CDX-001 | Replace invalid `approvalMode` with closest valid | unsafe |
| CDX-002 | Replace invalid `fullAutoErrorMode` with closest valid | unsafe |
| OC-001 | Replace invalid `share` mode with closest valid | unsafe |
| CC-HK-016 | Replace unknown hook `type` with closest valid | unsafe |

### Key changes

- **New shared helper**: `find_unique_json_string_value_span` consolidated in `rules/mod.rs` (was duplicated in mcp.rs)
- **New TOML helper**: `find_toml_string_value_span` in codex.rs for TOML config files
- **Fuzzy matching**: CDX-001, CDX-002, OC-001, CC-HK-016 use `find_closest_value` for smart replacement suggestions
- **Uniqueness guard**: All span finders use iterator-based uniqueness checks to prevent ambiguous fixes
- Updated `rules.json`, `VALIDATION-RULES.md`, and `locales/en.yml` for all 8 rules

### Test coverage

- 21+ new tests covering: fix safety, replacement values, byte targeting, uniqueness guards, edge cases (empty strings, type mismatches, non-string types, duplicate keys)
- All 2,273 tests passing

## Test plan

- [x] `cargo check` passes
- [x] `cargo test` — 2,273 tests pass, 0 failures
- [x] `cargo clippy -- -D warnings` — clean
- [x] `cargo build --release` — successful
- [x] Each rule has autofix tests verifying `fix.safe == false` and correct replacement
- [x] Uniqueness guard tests verify no fix when values are ambiguous

Closes #317

### CODE REVIEW

File: crates/agnix-core/src/rules/codex.rs [hunk 1]
```diff
@@ -7,14 +7,40 @@
...
+/// Returns byte positions of the inner string (without quotes).
+/// Returns None if the key is not found or appears more than once (uniqueness guard).
+fn find_toml_string_value_span(
+    content: &str,
+    key: &str,
+    current_value: &str,
+) -> Option<(usize, usize)> {
+    let pattern = format!(
+        r#"(?m){}\s*=\s*"({})"#,
+        regex::escape(key),
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/codex.rs:31]: Fixed - anchored the regex to line boundary with (?:^|\n)\s* so it won't match myapprovalMode as a substring. All 35 CDX tests pass.

File: crates/agnix-core/src/rules/codex.rs [hunk 2]
```diff
@@ -7,14 +7,40 @@
...
+        regex::escape(current_value)
+    );
+    let re = Regex::new(&pattern).ok()?;
+    let mut iter = re.captures_iter(content);
+    let first = iter.next()?;
+    if iter.next().is_some() {
+        return None; // Not unique
+    }
+    let m = first.get(1)?;
+    Some((m.start(), m.end()))
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/codex.rs:41]: Fixed - anchored the TOML regex to line start with (?:^|\n)\s* to prevent substring key matches. Also removed the unnecessary (?m) flag since we now use (?:^|\n) directly.

File: crates/agnix-core/src/rules/mod.rs [hunk 3]
```diff
@@ -120,6 +121,29 @@ pub(crate) fn find_yaml_value_range<T: FrontmatterRanges>(
...
+/// Find the byte span of a JSON string value for a unique key/value pair.
+/// Returns byte positions of the inner string (without quotes).
+/// Returns None if the key/value pair is not found or appears more than once (uniqueness guard).
+pub(crate) fn find_unique_json_string_value_span(
+    content: &str,
+    key: &str,
+    current_value: &str,
+) -> Option<(usize, usize)> {
+    let pattern = format!(
+        r#""{}"\s*:\s*"({})""#,
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/mod.rs:133]: This is a false positive due to Rust raw string syntax confusion. The pattern r#""{}"\s*:\s*"({})""# uses raw strings (r#"..."#) where inner double quotes are literal. The actual regex produced is "key"\s*:\s*"(value)" which correctly matches JSON "key": "value". No change needed.

File: crates/agnix-core/src/rules/codex.rs [hunk 4]
```diff
@@ -7,14 +7,40 @@
...
+/// Find the byte span of a TOML string value for the given key.
+/// Returns byte positions of the inner string (without quotes).
+/// Returns None if the key is not found or appears more than once (uniqueness guard).
+fn find_toml_string_value_span(
+    content: &str,
+    key: &str,
+    current_value: &str,
+) -> Option<(usize, usize)> {
+    let pattern = format!(
+        r#"(?m){}\s*=\s*"({})"#,
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/codex.rs]: Good catch - removed the unnecessary (?m) flag. The TOML regex now uses (?:^|\n)\s* for line anchoring directly instead of relying on multiline mode.
